### PR TITLE
feat: change target type of metric to Utilization

### PR DESCRIPTION
**Pull request by pommelinho**
## Checklist

- [ ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [ ] I have read the [security policy](../security/policy).
- [ ] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [ ] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added necessary documentation within the code base (if
      appropriate).

### CODE REVIEW

File: helm/charts/hydra/values.yaml [hunk 1]
```diff
@@ -165,8 +165,8 @@ deployment:
     enabled: false
   # minReplicas: 1
   # maxReplicas: 3
-  # cpuAverageValue: 200m
-  # memoryAverageValue: 0.5Gi
+  # cpuAverageUtilization: 80
+  # memoryAverageUtilization: 80
```
Comment (m1pl) [inline helm/charts/hydra/values.yaml:169]: Maybe rename them to ```targetCPUUtilizationPercentage``` and ```targetMemoryUtilizationPercentage```, as in the Helm chart scaffold?

File: helm/charts/hydra/templates/hpa.yaml [hunk 2]
```diff
@@ -14,20 +14,20 @@ spec:
...
-  {{- if .Values.deployment.autoscaling.cpuAverageValue }}
+  {{- if .Values.deployment.autoscaling.cpuAverageUtilization }}
     - type: Resource
       resource:
         name: cpu
         target: 
-          type: AverageValue 
-          averageValue: {{ .Values.deployment.autoscaling.cpuAverageValue }}
+          type: Utilization
+          averageUtilization: {{ .Values.deployment.autoscaling.cpuAverageUtilization }}
```
Comment (m1pl) [inline helm/charts/hydra/templates/hpa.yaml:23]: You can replace these 3 lines with: ```targetAverageUtilization: {{...}}```

File: helm/charts/hydra/templates/hpa.yaml [hunk 3]
```diff
@@ -14,20 +14,20 @@ spec:
...
-  {{- if .Values.deployment.autoscaling.memoryAverageValue }}
+  {{- if .Values.deployment.autoscaling.memoryAverageUtilization }}
     - type: Resource
       resource:
         name: memory
         target:
-          type: AverageValue
-          averageValue: {{ .Values.deployment.autoscaling.memoryAverageValue }}
+          type: Utilization
+          averageUtilization: {{ .Values.deployment.autoscaling.memoryAverageUtilization }}
```
Comment (m1pl) [inline helm/charts/hydra/templates/hpa.yaml:31]: Same here: ```targetAverageUtilization: {{...}}```

**Review by m1pl — CHANGES_REQUESTED** [review]

File: helm/charts/hydra/templates/hpa.yaml (hunk 2 above)
Comment (PeteMac88) [reply helm/charts/hydra/templates/hpa.yaml:23]: I think this holds for the `autoscaling/v2beta1` hpa but here its the `autoscaling/v2beta2`. I think the spec changed. See https://godoc.org/k8s.io/api/autoscaling/v2beta2#MetricTarget

**Review by PeteMac88 — CHANGES_REQUESTED** [review]
When doing it like this I think you need to switch the apiVersion to `v2beta1`

**Comment by pommelinho** [comment]
ok but v2beta2 is newer, right. i would prefer having a newer version

**Review by m1pl — APPROVED** [review]

**Comment by m1pl** [comment]
@pommelinho v2beta2 can use external metrics or define a scaling behaviour (how fast should it scale up/down). v2beta1 is also deprecated in Kubernetes 1.19, so yeah, we should update our charts not later than sometime next year, although it currently has no impact on us.

**Pull request merged**